src/bundler.zig

```diff
@@ -374,14 +374,21 @@ pub const Bundler = struct {
 
     fn _resolveEntryPoint(bundler: *Bundler, entry_point: string) !_resolver.Result {
         return bundler.resolver.resolve(bundler.fs.top_level_dir, entry_point, .entry_point) catch |err| {
-            if (!std.fs.path.isAbsolute(entry_point) and !strings.hasPrefix(entry_point, "./")) brk: {
-                return bundler.resolver.resolve(
-                    bundler.fs.top_level_dir,
-                    try strings.append(bundler.allocator, "./", entry_point),
-                    .entry_point,
-                ) catch
-                // return the original error
-                    break :brk;
+            // Relative entry points that were not resolved to a node_modules package are
+            // interpreted as relative to the current working directory.
+            if (!std.fs.path.isAbsolute(entry_point) and
+                !(strings.hasPrefix(entry_point, "./") or strings.hasPrefix(entry_point, ".\\")))
```

Comment:
Bun no longer allows `bun build /src/hello.js` to resolve to `./src/hello.js`. Even with outbase this doesnt make sense, and it matches ESBuild's behavior. 
See test `edgecase/AbsolutePathShouldNotResolveAsRelative` which was previously `todo`.

Still allow `bun build src/hello.js` to resolve to `./src/hello.js` which is what the changed code is supposed to do